modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/dag/schema.scala

```diff
@@ -11,19 +11,42 @@ object schema {
 
   case class GenericMetadata()
 
+  @derive(eqv, show)
+  sealed trait SignatureType extends EnumEntry
+
+  object SignatureType extends Enum[SignatureType] with SignatureTypeNamer {
+    val values = findValues
+
+    case object Ecdsa extends SignatureType
+    case object EcdsaRecovery extends SignatureType
+    case object Ed25519 extends SignatureType
+    case object Schnorr1 extends SignatureType
+    case object SchnorrPoseidon extends SignatureType
+  }
+
+  trait SignatureTypeNamer {
+    implicit val name: Encoder[SignatureType] =
+      Encoder.forProduct1("status")(_ match {
```

Comment:
Field name is still unchanged

Reply:
I get what you are trying to say now. Changed to match `NodeState`.